Source/XUnitExtensions/AssertWithDiff.cs

```diff
@@ -0,0 +1,37 @@
+using System.Text;
+using DiffPlex;
+using DiffPlex.DiffBuilder;
+using DiffPlex.DiffBuilder.Model;
+using Xunit.Sdk;
+
+namespace XUnitExtensions {
+  public class AssertWithDiff {
+    public static void Equal(string expected, string actual) {
+      var diff = InlineDiffBuilder.Instance.BuildDiffModel(expected, actual);
+      if (!diff.HasDifferences) {
+        return;
+      }
+
+      var message = new StringBuilder();
+      message.AppendLine("AssertEqualWithDiff() Failure");
+      message.AppendLine("Diff (changing expected into actual):");
+      foreach (var line in diff.Lines) {
+        switch (line.Type) {
+          case ChangeType.Inserted:
+            message.Append('+');
```

Comment:
Do we want `{}` here?

Reply:
The goal of this text is to be appended, so I understand why @robin-aws  did not concatenate it.
However, I simplified the code a bit with a switch expression:

```
        var prefix = line.Type switch
        {
          ChangeType.Inserted => '+',
          ChangeType.Deleted => '-',
          _ => ' '
        };
        message.Append(prefix);
```